server/src/main/java/org/elasticsearch/index/codec/vectors/IVFVectorsReader.java

```diff
@@ -225,8 +225,8 @@ public final void search(String field, float[] target, KnnCollector knnCollector
         }
         int numVectors = rawVectorsReader.getFloatVectorValues(field).size();
         BitSet visitedDocs = new FixedBitSet(state.segmentInfo.maxDoc() + 1);
```

Comment:
I don't think this BitSet is needed anymore?

Reply:
@iverase I simplified things, and am now never checking for duplicates during initial search. This also significantly sped up filtered search (as much as 2x improvement!).